From 7662587f10552d4aad04ba2b95e5643ae2107d6f Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/installer/pull/15039
From: Antoine Martin <dev@ayakael.net>
Date: Wed, 19 Oct 2022 15:21:40 +0200
Subject: [PATCH 1/1] Enabled source-building with mono runtime on any
 architecture.

Arm does not support NativeAot, thus we have to set
'NativeAotSupported' when building on arm.

diff --git a/repos/sdk.proj b/repos/sdk.proj
index c18e00dd61..6567b41983 100644
--- a/repos/sdk.proj
+++ b/repos/sdk.proj
@@ -6,6 +6,7 @@
     <BuildCommandArgs>$(BuildCommandArgs) $(FlagParameterPrefix)nodereuse $(ArcadeFalseBoolBuildArg)</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:PackageProjectUrl=https://github.com/dotnet/sdk</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:PublishCompressedFilesPathPrefix=$(SourceBuiltToolsetDir)</BuildCommandArgs>
+    <BuildCommandArgs Condition="'$(BuildArchitecture)' == 'arm'">$(BuildCommandArgs) /p:NativeAotSupported=false</BuildCommandArgs>
 
     <LogVerbosityOptOut>true</LogVerbosityOptOut>
     <BuildCommandArgs>$(BuildCommandArgs) -v $(LogVerbosity)</BuildCommandArgs>
-- 
2.36.3

